matrix-sdk-android/src/main/java/org/matrix/android/sdk/api/session/permalinks/PermalinkParser.kt

```diff
@@ -65,17 +68,58 @@ object PermalinkParser {
                 PermalinkData.RoomLink(
                         roomIdOrAlias = identifier,
                         isRoomAlias = false,
-                        eventId = extraParameter.takeIf { !it.isNullOrEmpty() && MatrixPatterns.isEventId(it) }
+                        eventId = extraParameter.takeIf { !it.isNullOrEmpty() && MatrixPatterns.isEventId(it) },
+                        viaParameters = viaQueryParameters
                 )
             }
             MatrixPatterns.isRoomAlias(identifier) -> {
                 PermalinkData.RoomLink(
                         roomIdOrAlias = identifier,
                         isRoomAlias = true,
-                        eventId = extraParameter.takeIf { !it.isNullOrEmpty() && MatrixPatterns.isEventId(it) }
+                        eventId = extraParameter.takeIf { !it.isNullOrEmpty() && MatrixPatterns.isEventId(it) },
+                        viaParameters = viaQueryParameters
                 )
             }
             else                                   -> PermalinkData.FallbackLink(uri)
         }
     }
+
+    private fun String.getViaParameters(indexOfQuery: Int): List<String> {
+        val query = try {
+            substring(indexOfQuery + 1)
+        } catch (e: IndexOutOfBoundsException) {
+            return emptyList()
+        }
+        val encodedKey = try {
+            URLEncoder.encode("via", "UTF-8")
```

Comment:
Is it really necessary? the result will always be "via"...

Reply:
Ooops, sorry, was a copy paste from Uri code